src/openms/source/DATASTRUCTURES/QTCluster.cpp

```diff
@@ -182,47 +190,43 @@ namespace OpenMS
     }
   }
 
-  void QTCluster::getElements(OpenMSBoost::unordered_map<Size, OpenMS::GridFeature*>& elements)
+  QTCluster::ClusterElementsMap QTCluster::getElements() const
   {
     OPENMS_PRECONDITION(finalized_,
         "Cannot perform operation on cluster that is not finalized")
 
-    elements.clear();
-    elements[center_point_->getMapIndex()] = center_point_;
+    // copy neighbor map
+    NeighborMap neighbors_copy = data_->neighbors_;
 
-    if (neighbors_.empty())
-    {
-      return;
-    }
+    // add center point to the copy
+    neighbors_copy[data_->center_point_->getMapIndex()].second = data_->center_point_;
 
-    // since we are finalized, we do not need to care about the annotation
-    for (NeighborMap::const_iterator it = neighbors_.begin(); it != neighbors_.end(); ++it)
-    {
-      elements[it->first] = it->second.second;
-    }
+    return neighbors_copy;
```

Comment:
I guess named return value optimization (NRVO) is in effect here... should we add a comment?

Reply:
Done.